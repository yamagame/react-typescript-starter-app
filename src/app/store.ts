import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import articleReducer from '../features/article/articleSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    article: articleReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
