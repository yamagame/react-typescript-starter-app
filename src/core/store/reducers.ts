import counterReducer from 'core/slices/counter';
import articleReducer from 'core/slices/article';
import formSampleReducer from 'core/slices/formSample';
import todoReducer from 'core/slices/todo';

export const reducer = {
  counter: counterReducer,
  article: articleReducer,
  formSample: formSampleReducer,
  todo: todoReducer,
};
