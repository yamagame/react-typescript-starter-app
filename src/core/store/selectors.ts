import { RootState } from '.';

export const selectors = {
  article: (state: RootState) => state.article,
  counter: (state: RootState) => state.counter,
  formSample: (state: RootState) => state.formSample,
  todo: (state: RootState) => state.todo,
};
