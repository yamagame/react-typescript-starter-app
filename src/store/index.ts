import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

import applicatoinReducer from 'features/application';
import counterReducer from 'features/counter';
import articleReducer from 'features/article';
import formSampleReducer from 'features/formSample';
import todoReducer from 'features/todo';

export const store = configureStore({
  reducer: {
    application: applicatoinReducer,
    counter: counterReducer,
    article: articleReducer,
    formSample: formSampleReducer,
    todo: todoReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const selectors = {
  application: (state: RootState) => state.application,
  article: (state: RootState) => state.article,
  counter: (state: RootState) => state.counter,
  formSample: (state: RootState) => state.formSample,
  todo: (state: RootState) => state.todo,
};
