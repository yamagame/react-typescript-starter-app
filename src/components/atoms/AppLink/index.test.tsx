import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from 'store';
import { AppLink } from '.';

test('AppLinkの表示', () => {
  const { container } = render(
    <Provider store={store}>
      <AppLink href="string" name="string" />
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
