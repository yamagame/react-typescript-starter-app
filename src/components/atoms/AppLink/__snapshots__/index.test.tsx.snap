// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLinkの表示 1`] = `
<div>
  <a
    class="App-link"
    href="string"
    rel="noopener noreferrer"
    target="_blank"
  >
    string
  </a>
</div>
`;
